internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,699 @@
+                    VALUE_STEP => item.accessible_string_property(AccessibleStringProperty::ValueStep),
+                    _ => Default::default(),
+                };
+                QString::from(string.as_ref())
+            } else {
+                QString::default()
+            }
+        });
+    }
+
+    // ------------------------------------------------------------------------------
+    // Slint_accessible:
+    // ------------------------------------------------------------------------------
+
+    // Base object for accessibility support
+    class Slint_accessible : public QAccessibleInterface {
+    public:
+        Slint_accessible(QObject *obj, QAccessible::Role role, QAccessibleInterface *parent) :
+             has_focus(false), has_focus_delegation(false), m_role(role), m_parent(parent), m_object(obj)
+        { }
+
+        ~Slint_accessible() {
+            qDeleteAll(m_children);
+        }
+
+        virtual void *rustItem() const = 0;
+
+        // Returns the SlintWidget of the window... we have no other.
+        virtual QWidget *qwidget() const = 0;
+
+        QPoint mapToGlobal(const QPoint p) const {
+            return qwidget()->mapToGlobal(p);
+        }
+
+        QPoint mapFromGlobal(const QPoint p) const {
+            return qwidget()->mapFromGlobal(p);
+        }
+
+        void clearFocus() {
+            has_focus = false;
+            has_focus_delegation = false;
+
+            for (int i = 0; i < childCount(); ++i) {
+                static_cast<Slint_accessible *>(child(i))->clearFocus();
+            }
+        }
+
+        virtual void delegateFocus() const {
+            sendFocusChangeEvent();
+        }
+
+        bool focusItem(void *item) const {
+            auto my_item = rustItem();
+            if (rust!(Slint_accessible_findItem [item: &ItemWeak as "void *", my_item: &ItemWeak as "void*"] -> bool as "bool" {
+                item == my_item
+            })) {
+                has_focus = true;
+
+                delegateFocus();
+                return true;
+            }
+
+            for (int i = 0; i < childCount(); ++i) {
+                if (static_cast<Slint_accessible *>(child(i))->focusItem(item)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        void sendFocusChangeEvent() const {
+            auto event = QAccessibleEvent(object(), QAccessible::Focus);
+            QAccessible::updateAccessibility(&event);
+            has_focus_delegation = true;
+        }
+
+        bool isValid() const override {
+            return true;
+        }
+
+        QObject *object() const override {
+            return m_object;
+        }
+
+        // relations
+        QVector<QPair<QAccessibleInterface *, QAccessible::Relation>>
+        relations(QAccessible::Relation match = QAccessible::AllRelations) const override {
+            Q_UNUSED(match);
+            return {}; /* FIXME */
+        }
```

Comment:
I'm curious, is there a reason for re-implementing this in the same way as the default implementation behaves?

Reply:
I copied the entire interface into my code marked with FIXME to make sure I do not forget something.The default implementation is fine, I'll just remove this again.